psi4/src/psi4/libscf_solver/hf.cc

```diff
@@ -705,10 +714,18 @@ void HF::form_Shalf() {
 
     // Convert the eigenvales to 1/sqrt(eigenvalues)
     const Dimension& dimpi = eigval->dimpi();
-    double min_S = std::fabs(eigval->get(0, 0));
+    // Cannot assume that (0,0) is a valid reference
+    bool min_S_initialized = false;
+    double min_S;
     for (int h = 0; h < nirrep_; ++h) {
         for (int i = 0; i < dimpi[h]; ++i) {
-            if (min_S > eigval->get(h, i)) min_S = eigval->get(h, i);
+            if (!min_S_initialized) {
+                min_S = eigval->get(h, i);
```

Comment:
I'm wondering, above we used to initialise using `abs(eigval[0,0])`, here it's just `eigval[h,i]`. Is that really correct? I guess values in `eigval` must be positive, otherwise the `sqrt(eigval[h,i])` below wouldn't work, but I'm just checking.

Reply:
Good catch. The old version was actually wrong. In the theoretical case (I'm not sure if it even exists?) where you would have a negative eigenvalue but no eigenvalues close to zero, the old version would have just used symmetric orthogonalization, whereas the correct way is, of course, to only take the positive definite part.

But, this is just an academic discussion: the overlap matrix is supposed to be positive definite. Sometimes you do see small negative eigenvalues.

Reply:
Also note that the square root is only taken from eigenvalues that are bigger than the cutoff that is positive.